lib/run.bash

```diff
@@ -23,19 +23,34 @@ compose_cleanup() {
 check_linked_containers_and_save_logs() {
   local service="$1"
   local logdir="$2"
+  local uploadlogs="$3"
+  local uploadall="false"
+
+  if [[ "$uploadlogs" =~ ^(false|off|0|never)$ ]]; then
+    # Skip all if we are not uploading logs
+    return
+  elif [[ "$uploadlogs" =~ ^(true|on|1|always)$ ]]; then
+    uploadall="true"
+  fi
 
   [[ -d "$logdir" ]] && rm -rf "$logdir"
   mkdir -p "$logdir"
 
-  while read -r line ; do
-    if [[ -z "${line}" ]]; then
+  # Get array of containers
+  containers=$(docker_ps_by_project --format '{{.ID}}\t{{.Label "com.docker.compose.service"}}')
+
+  IFS=$'\n'      # Change IFS to new line
```

Comment:
I had to rework the `while` to this `for` with `IFS` to enable testing, otherwise `BATS` eats the `stdin` to the loop and basically kills the rest of the testing.